src/LineBot/Events/ILineEvents.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright 2017-2018 Dirk Lemstra (https://github.com/dlemstra/line-bot-sdk-dotnet)
+//
+// Dirk Lemstra licenses this file to you under the Apache License,
+// version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at:
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+using System.Collections.Generic;
+
+namespace Line
+{
+    /// <summary>
+    /// Encapsulates the interfacce for a list of line events.
+    /// </summary>
+    public interface ILineEvents : IEnumerable<ILineEvent>
+    {
+        /// <summary>
+        /// Gets or sets all the events.
+        /// </summary>
+        List<ILineEvent> Events { get; set; }
```

Comment:
This should be read only. The interface can only have a getter while the implementation has both a setter and a getter.